src/main/java/appeng/client/me/ItemRepo.java

```diff
@@ -111,14 +114,25 @@ public void updateView() {
         }
 
         this.innerSearch = this.searchString;
-        final boolean terminalSearchToolTips =
-                AEConfig.instance.settings.getSetting(Settings.SEARCH_TOOLTIPS) != YesNo.NO;
+        //        final boolean terminalSearchToolTips =
+        //                AEConfig.instance.settings.getSetting(Settings.SEARCH_TOOLTIPS) != YesNo.NO;
         // boolean terminalSearchMods = Configuration.INSTANCE.settings.getSetting( Settings.SEARCH_MODS ) != YesNo.NO;
 
         boolean searchMod = false;
-        if (this.innerSearch.startsWith("@")) {
+        boolean searchTooltips = false;
```

Comment:
The idea is nice, but since now there are so many modes maybe introduce  an enum & use switch to get a bit cleaner code?

Reply:
fixed

Reply:
This version is much better but I still think a separate enum listing search modes would make the second `elseif` ladder look better. I don't insist though. (One possible benefit of an enum is that it may incapsulate this switch in a member method.)